### ISSUE TITLE
Accessibility: Disabling of submit buttons should be avoided for screen readers

### ISSUE THREAD

**Issue by philippsauer** [intershop/intershop-pwa#1836]
**Actual Behavior**

In some forms of the Intershop PWA, submit buttons are disabled (examples below). 

Disabled submit buttons are problematic because they are not focusable and not discoverable for screen readers. Also, once disabled, they prevent users who are trying to submit the form from receiving feedback.

For example:
* Quick order form on cart page: SR users won't even know that there is a submit button
* Registration form: After first activation, it will move the focus to the first invalid field (OK), but this should not disable the button so this action can be repeated if necessary.

**Expected Behavior**

- The Submit button should generally remain enabled and focusable, even if form fields are invalid. The submit button should always be part of the tab order (tabindex="0") and clickable by keyboard and mouse users. (Can be marked with aria-disabled=”true”)
- If the user attempts to submit the form while there are invalid fields: Prevent the form from submitting. Ensure all relevant error messages are prominently displayed and move keyboard focus to the first invalid field (as already implemented) or to a summary of all errors at the top of the form, allowing screen reader users to easily find what needs fixing.

**Additional Context, like Screenshots, Log File Snippets etc.**

Disabling submit buttons using the native disabled attribute removes them from the accessibility tree and tab order, making them invisible and inaccessible to screen reader users and keyboard-only users. This violates WCAG 2.1 Success Criterion 2.1.1 (Keyboard) and 4.1.2 (Name, Role, Value).

[AB#107476](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/107476)
